kod/object/passive/skill/disarm.kod

```diff
@@ -140,7 +140,17 @@ messages:
          {
             % Currently, have 1/5th your percent plus your modifier chance to make opponent drop weapon.
             iAbil = (iAbil + 1 + iModifier) / DISARM_FACTOR;
-            if (random(1,100) < iAbil) and (not Send(Send(oTarget,@GetOwner),@IsArena))
+
+            % Cannot force a weapon to drop if...
+            % ...the target is a DM (generally, an event target)
+            % ...the target is in an arena
+            % ...the target's weapon doesn't drop on death (soft white light/transcendent)
+            % ...the target's weapon shouldn't leave them for other reasons
+            if (Random(1, 100) < iAbil)
+               AND NOT IsClass(oTarget, &DM)
```

Comment:
Hm, these kind of checks are fragile.  e.g. if someone ever added a new kind of special room that was like an arena but wasn't IsArena, there would be hundreds of places to edit, which shows that the design isn't right.

So. @ReqLeaveOwner: yes, definitely, this is intended usage.
@IsClass(DM): unpleasant use of IsClass, but I don't think there's currently a better alternative.  It would be better if we had a @CanBeAffectedByMortals or something, but I think we don't.

As for the other two, they're kind of gross, and 4 checks certainly feels like too many.  Maybe there's no other way around it, but can you do some digging and make sure there's not a better existing check?

Reply:
- @IsArena: I didn't find anything more fitting as a substitute for this.  The ROOM_SAFE_DEATH flag is the closest alternative, but that seems even worse.  Plus, after some further digging, IsArena is used similarly by several other spells/skills to restrict behaviors in various ways, so it might be the best option in terms of consistency.  Players have come to accept that arenas change the rules for several spells and abilities already.
- @DropOnDeath, IsClass DM: Would a cleaner approach be to add something like a pbCanDropWeaponOnDisarm property to battler.kod and CanDropWeaponOnDisarm() to battler.kod and itematt.kod?  Normally TRUE for players and itematts, DMs and the Soft-White-Light item attribute would override this with FALSE. (Plus, DMs could set their property to TRUE should they want that functionality to exist for an event or something)

The result would now require this set of checks to all be TRUE for disarm to drop the target's weapon to the ground:  Random 1-100 < iAbil, ReqLeaveOwner, IsArena, CanDropWeaponOnDisarm

Definitely want your thumbs-up or thumbs-down before I proceed with this, though.